### PR TITLE
Move common dedup utils and remove unused code

### PR DESCRIPTION
**Pull request by ayushdg**
This PR remove the `nemo_curator/gpu_deduplication` folder in favor of using all code from either the fuzzy_dedup module or fuzzy_dedup_utils. (A few methods were left behind).

It also moves the fuzzy dedup scripts into a new subfolder with a readme on the order of execution and example usage.
It adds a caution to the gpu_deduplication slurm example currently in `examples` which will be removed in a followup and replaced by a python only API example.

- [X] Verified all tests passing
- [X] Verified refactored scripts run w/ expected results locally

### CODE REVIEW

File: nemo_curator/scripts/fuzzy_deduplication/legacy/verify_all_pairs_jaccard.py
Comment (ryantwolf) [inline nemo_curator/scripts/fuzzy_deduplication/legacy/verify_all_pairs_jaccard.py:1]: Why are we keeping these files around? They use some files that no longer exist like:
```
nemo_curator.gpu_deduplication.jaccard_utils.jaccard_similarity_utils
```

File: nemo_curator/scripts/fuzzy_deduplication/README.md
Comment (ryantwolf) [inline nemo_curator/scripts/fuzzy_deduplication/README.md:1]: I like this documentation. Could you include it in the user guide too? [Here](https://github.com/NVIDIA/NeMo-Curator/blob/main/docs/user-guide/GpuDeduplication.rst) in particular.

File: setup.py [hunk 1]
```diff
@@ -89,18 +89,13 @@
...
             "remove_matching_ngrams=nemo_curator.scripts.remove_matching_ngrams:console_script",
-            "gpu_compute_minhashes=nemo_curator.scripts.compute_minhashes:console_script",
-            "minhash_buckets=nemo_curator.scripts.minhash_lsh:console_script",
-            "jaccard_map_buckets=nemo_curator.scripts.map_buckets:console_script",
-            "jaccard_shuffle=nemo_curator.scripts.jaccard_shuffle:console_script",
-            "jaccard_compute=nemo_curator.scripts.jaccard_compute:console_script",
-            "gpu_connected_component=nemo_curator.scripts.connected_components:console_script",
-            "write_deduped_result_with_text=nemo_curator.gpu_deduplication.write_deduped_result_with_text:console_script",
-            "verify_all_pairs_jaccard=nemo_curator.gpu_deduplication.verify_all_pairs_jaccard:console_script",
+            "gpu_compute_minhashes=nemo_curator.scripts.fuzzy_deduplication.compute_minhashes:console_script",
```
Comment (ryantwolf) [inline setup.py:92]: I get this error when trying to run the scripts:
```
Traceback (most recent call last):
  File "/home/rywolf/.local/bin/minhash_buckets", line 5, in <module>
    from nemo_curator.scripts.fuzzy_deduplication.minhash_lsh import console_script
ModuleNotFoundError: No module named 'nemo_curator.scripts.fuzzy_deduplication'
```
I think you need an `__init__.py` in the new subfolder you created.

**Review by ryantwolf — CHANGES_REQUESTED** [review]
Found one bug. I think it looks pretty good otherwise.

File: nemo_curator/scripts/fuzzy_deduplication/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+## Fuzzy Deduplication Steps
+This directory consists of scripts that can be invoked directly via the command line for finding fuzzy duplicates from a group of Jsonl files consisting of text & unique ID's that are specifically formatted using the `add_id` script included as a part of NeMo-Curator.
+
+> [!IMPORTANT]
+> The scripts are helper utilities that wrap the fuzzy_dedup API for handling multiple jsonl directories and the id format generated by [add_id](../add_id.py). For most cases we recommend working with the fuzzy_deduplication API directly.
+
```
Comment (VibhuJawa) [inline nemo_curator/scripts/fuzzy_deduplication/README.md:6]: Love the documentation here.

**Review by VibhuJawa — APPROVED** [review]
Looks good to me (as long as we address Ryan's changes) .  Thanks for the code cleanup and documentation.

File: nemo_curator/scripts/fuzzy_deduplication/legacy/verify_all_pairs_jaccard.py
Comment (ayushdg) [reply nemo_curator/scripts/fuzzy_deduplication/legacy/verify_all_pairs_jaccard.py:1]: Makes sense removed.

**Review by ryantwolf — APPROVED** [review]
All my concerns were addressed. Thanks!

**Pull request merged**